src/AzureClient/AzureClient.cs

```diff
@@ -572,10 +585,31 @@ public async Task<ExecutionResult> GetJobResultAsync(IChannel? channel, string j
                 //                 cancellation token support.
                 var request = WebRequest.Create(job.OutputDataUri);
                 using var responseStream = (await request.GetResponseAsync()).GetResponseStream();
-                return responseStream.ToHistogram(
-                    Logger, 
-                    isSimulatorOutput: this.ActiveTarget?.TargetId?.StartsWith(MicrosoftSimulator) ?? false)
-                .ToExecutionResult();
+                if (this.ActiveTarget?.TargetId?.StartsWith(MicrosoftSimulator) ?? false)
+                {
+                    var outputLines = ReadAllLines(responseStream).ToList();
+                    outputLines = outputLines.Select(line => line.Trim()).ToList();
+                    
+                    // ToDo: This is a hack to separate Messages from the result.
```

Comment:
```suggestion
                    // N.B. The current simulator output format is just text and it does not distinguish between the result of the operation and other kinds of output.
                    //      Attempt to parse the output to distinguish the result from the rest of the output until the simulator output format makes it easy to do so.
```

Reply:
Expanded the comment to provide more context on why the custom parsing is needed.

Reply:
done